### ISSUE TITLE
Modal: add aria-label to dialog element (2.4.6, 4.1.2)

### ISSUE THREAD

**Issue by Blackbaud-MattGregg** [blackbaud/skyux1#962]
### Expected behavior
The dialog has an explicit label programmatically connected to it. 

This could be done by pointing an aria-labelledBy attribute on the role="dialog" element to the id of the modal's header but teams would need to make this connection on a case-by-case basis to ensure they are unique. 

The approach discussed would be to use the aria-label attribute and have it's value be the same as the modal's title displayed in the header. This needs to be added to the component. 

https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal
